packages/pluggableWidgets/color-picker-web/src/components/ColorPicker.tsx

```diff
@@ -0,0 +1,152 @@
+    | SketchPickerProps
+    | ChromePickerProps
+    | TwitterPickerProps
+    | HuePickerProps
+    | BlockPickerProps
+    | GithubPickerProps
+    | CirclePickerProps
+    | SwatchesPickerProps
+    | SliderPickerProps
+    | MaterialPickerProps
+    | CompactPickerProps;
+export const ColorPicker = (props: ColorPickerProps): ReactElement => {
+    const defaultColor = {
+        hex: "#FFFFFF",
+        rgb: "rgb(255,255,255)",
+        rgba: "rgb(255,255,255,1)"
+    };
+    const { type, mode, disabled, defaultColors, color, format, invalidFormatMessage } = props;
+    const colorElement = getColorPicker(type);
+    const [hidden, setHidden] = useState(mode !== "inline");
+    const [config, setConfig] = useState<ColorPickerConfigProps>({});
+    const [currentColor, setCurrentColor] = useState<string>(color || defaultColor[format]);
+    const [alertMessage, setAlertMessage] = useState<string | undefined>();
+
+    const validateColor = (color: string): void => {
+        const validFormat = validateColorFormat(color, format);
+        const alertMessage = validFormat ? invalidFormatMessage?.replace(/\{1}/, validFormat) : undefined;
+        setAlertMessage(alertMessage);
+    };
+    const onToggle = useCallback(
+        (hide: boolean): void => {
+            if (!disabled && mode !== "inline") {
+                setHidden(hide);
+            }
+        },
+        [disabled, mode]
+    );
+    const onColorValueUpdate = useCallback(
+        (color: ColorState): void => {
+            if (color && !disabled) {
+                const finalColor = parseColor(color, format);
+                setCurrentColor(finalColor);
+                props.onChangeComplete(finalColor);
+            }
+        },
+        [disabled, format, props]
+    );
+
+    const onInputChange = (event: React.ChangeEvent<HTMLInputElement>): void => {
+        const value = event.target.value;
+        validateColor(color);
+        setCurrentColor(value);
+        props.onChangeComplete(value);
+    };
+
+    const renderInput = (): ReactElement => {
+        return (
+            <Input color={color} disabled={disabled} onKeyUp={onToggle} onChange={onInputChange}>
+                {renderButton()}
+            </Input>
+        );
+    };
+    const renderButton = (): ReactElement => {
+        return <Button mode={mode} hidden={hidden} disabled={disabled} onClick={onToggle} color={color} />;
+    };
+    const renderColorPicker = (): ReactElement => {
+        const supportPopover = mode !== "inline" && type !== "hue" && type !== "slider";
+        return (
+            <div
+                className={classNames({
+                    "widget-color-picker-popover": supportPopover,
+                    "widget-color-picker-no-popover": mode !== "inline"
+                })}
+            >
+                {mode !== "inline" ? (
+                    <div className={"widget-color-picker-cover"} onClick={() => onToggle(true)} />
```

Comment:
From what I understand, this div is always shown but only sets the `hidden` state to `true`. Is that expected compared to e.g. not showing it when `hidden` is `true`?

Reply:
This div appears only when `ColorPicker` component is presented. 
So basically there is no need to add condition on its visibility, because when `ColorPicker` is hidden this div basically doesn't exist, when `ColorPicker` is not hidden then this div should be presented.